testbed/tests/trace_test.go

```diff
@@ -349,16 +405,139 @@ func TestTraceAttributesProcessor(t *testing.T) {
 			// Create a service name that does not match "include" filter.
 			nodeToExclude := &commonpb.Node{ServiceInfo: &commonpb.ServiceInfo{Name: "service-not-to-add-attr"}}
 
-			verifySingleSpan(t, tc, nodeToExclude, spanToInclude, func(span *tracepb.Span) {
+			verifySingleJaegerSpan(t, tc, nodeToExclude, spanToInclude, func(span *tracepb.Span) {
+				// Verify attributes was not added.
+				assert.Nil(t, span.Attributes)
+			})
+
+			// Create another span that does not match "include" filter.
+			spanToExclude := &tracepb.Span{
+				Name: &tracepb.TruncatableString{Value: "span-not-to-add-attr"},
+			}
+			verifySingleJaegerSpan(t, tc, nodeToInclude, spanToExclude, func(span *tracepb.Span) {
 				// Verify attributes was not added.
 				assert.Nil(t, span.Attributes)
 			})
+		})
+	}
+}
+
+// verifySingleZipkinSpan sends a single span to Collector, waits until the span is forwarded
+// and received by MockBackend and calls user-supplied verification function on
+// received span.
+func verifySingleZipkinSpan(
+	t *testing.T,
+	tc *testbed.TestCase,
+	span *tracepb.Span,
+	verifyReceived func(span *tracepb.Span),
+) {
+
+	// Clear previously received traces.
+	tc.MockBackend.ClearReceivedItems()
+	startCounter := tc.MockBackend.DataItemsReceived()
+
+	// Add dummy IDs
+	span.TraceId = testbed.GenerateTraceID(1)
+	span.SpanId = testbed.GenerateSpanID(1)
+	td := consumerdata.TraceData{Spans: []*tracepb.Span{span}}
+
+	sender := tc.Sender.(testbed.TraceDataSender)
+
+	// Send the span.
+	sender.SendSpans(td)
+
+	// We bypass the load generator in this test, but make sure to increment the
+	// counter since it is used in final reports.
+	tc.LoadGenerator.IncDataItemsSent()
+
+	// Wait until span is received.
+	tc.WaitFor(func() bool { return tc.MockBackend.DataItemsReceived() == startCounter+1 },
+		"Zipkin span received")
+
+	// Verify received span.
+	count := 0
+	for _, td := range tc.MockBackend.ReceivedTraces {
+		for _, span := range td.Spans {
+			verifyReceived(span)
+			count++
+		}
+	}
+	assert.EqualValues(t, 1, count, "must receive one Zipkin span")
+}
+
+func TestZipkinTraceAttributesProcessor(t *testing.T) {
```

Comment:
This should be possible to add to TestTraceAttributesProcessor as a subtest.

Reply:
@tigrannajaryan I've created a separated test because IMHO processor used in `TestTraceAttributesProcessor` was not working correctly for Zipkin. When I removed from include attributes the `services` key then the test was modifying traces and they were passing. Maybe I'm missing something about data models or something else?